lib/screens/v2/candidate_generator/widgets/elevator_closures.ex

```diff
@@ -20,27 +44,37 @@ defmodule Screens.V2.CandidateGenerator.Widgets.ElevatorClosures do
         fetch_elevator_alerts_with_facilities_fn \\ &Alert.fetch_elevator_alerts_with_facilities/0
       ) do
     with {:ok, location_context} <- fetch_location_context_fn.(PreFare, parent_station_id, now),
-         {:ok, parent_station_map} <- Stop.fetch_parent_station_name_map(),
+         {:ok, parent_station_map} <- @stop.fetch_parent_station_name_map(),
          {:ok, alerts} <- fetch_elevator_alerts_with_facilities_fn.() do
-      elevator_closures = relevant_alerts(alerts)
-      icon_map = get_icon_map(elevator_closures, parent_station_id)
+      elevator_closures =
+        alerts
+        |> elevator_alerts()
+        |> Enum.flat_map(&closure_details/1)
+
+      elevator_closure_ids =
+        elevator_closures
+        |> Enum.filter(&relevant_closure?(&1, elevator_closures, parent_station_id))
+        |> Enum.map(fn %{alert_id: alert_id} -> alert_id end)
+        |> MapSet.new()
 
       [
         %ElevatorStatusWidget{
-          alerts: elevator_closures,
+          alerts:
+            alerts
+            |> Enum.filter(fn %Alert{id: id} -> id in elevator_closure_ids end),
```

Comment:
No need for the pipeline here as it's only a single step:

```ex
Enum.filter(alerts, fn %Alert{id: id} -> id in elevator_closure_ids end)
```

Reply:
much cleaner, thanks!